src/helpers/algebra/cholesky.jl

```diff
@@ -43,9 +47,13 @@ function cholinv_logdet(x::AbstractMatrix{T}) where {T <: LinearAlgebra.BlasFloa
     # return inverse and log-determinant
     return y.factors, ly
 end
+
 cholinv_logdet(x::Diagonal) = Diagonal(inv.(diag(x))), mapreduce(z -> log(z), +, diag(x))
 cholinv_logdet(x::Real)     = inv(x), log(abs(x))
 
+fastcholesky(x::UniformScaling) = error("`fastcholesky` is not defined for `UniformScaling`. The shape is not determined.")
+fastcholesky!(x::UniformScaling) = error("`fastcholesky!` is not defined for `UniformScaling`. The shape is not determined.")
```

Comment:
Can't we do `fastcholesky(x::UniformScaling) = isposdef(x) ? Cholesky(sqrt(x.λ) * I, 'L', convert(BlasInt, 0)) : Error()`?

Reply:
Unfortunately no, because `factors` in the `Cholesky` structure must be of type `AbstractMatrix`, but `sqrt(x.λ) * I` is not an `AbstractMatrix`